core/src/consensus.rs

```diff
@@ -333,6 +346,28 @@ impl Tower {
         }
     }
 
+    pub fn aggregate_stake_lockouts(
+        &self,
+        ancestors: &HashMap<u64, HashSet<u64>>,
+        stake_lockouts: &HashMap<u64, StakeLockout>,
+    ) -> HashMap<u64, u128> {
+        let mut stake_weighted_lockouts: HashMap<u64, u128> = HashMap::new();
+        let root = self.root();
+        for (fork, lockout) in stake_lockouts.iter() {
```

Comment:
for forks that are no longer in `stake_lockouts`, either outdated `slot < root` or we decided to vote on a different fork 
for instance we went from 1-> 3 ->5 to 1->3->7, and '5' is no longer present, do these outdated banks get removed/cleaned up from the confidence cache. Do they need to be?

Reply:
Good call. I'm thinking `prune_non_root` might be an obvious place to do this.